src/react-hooks-nesting-walker/react-hooks-nesting-walker.ts

```diff
@@ -180,6 +180,21 @@ export class ReactHooksNestingWalker extends RuleWalker {
         return;
       }
 
+      /**
+       * Detect if the unnamed expression is wrapped in a illegal function call
+       */
+      if (
+        isCallExpression(ancestor.parent) &&
+        isIdentifier(ancestor.parent.expression) &&
+        !isReactComponentDecorator(ancestor.parent.expression)
```

Comment:
Looking at this logic, it looks similar to the condition in the next `if` below. Do you think we could refactor the code to avoid making the same checks twice?

Something along the lines of:

```ts
if (isCallExpression(ancestor.parent)) {
  if (!isReactComponentRecorator(ancestor.parent.expression)) {
    // failure
  }

  return;
}
```

Reply:
I actually don't like making the code complex by coupling conditionals w/ conditionals above. Precisely because we'd be trading simplicity and readability for a couple of expressions. But that code can be simplified further.